lib/auth/in_memory/backend.js

```diff
@@ -3,16 +3,37 @@
 const crypto = require('crypto');
 
 const errors = require('../../errors');
-const accountsKeyedbyAccessKey =
-    require('./vault.json').accountsKeyedbyAccessKey;
-const accountsKeyedbyCanID =
-    require('./vault.json').accountsKeyedbyCanID;
-const accountsKeyedbyEmail =
-    require('./vault.json').accountsKeyedbyEmail;
 const calculateSigningKey = require('./vaultUtilities').calculateSigningKey;
 const hashSignature = require('./vaultUtilities').hashSignature;
+const Indexer = require('./indexer');
+
+function _buildArn(generalResource, specificResource) {
+    return `arn:aws:s3:::${generalResource}/${specificResource}`;
```

Comment:
If you make service, here s3 in the arn configurable, this can be used by some other lightweight S3 we may build (e.g. Backbeat)